lib/cos_agent.py

```diff
@@ -0,0 +1,284 @@
+# Copyright 2025 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+r"""## Overview.
+This is a stripped-down version of cos_agent library provided by Grafana Agent
+charm (https://charmhub.io/grafana-agent/libraries/cos_agent). It removes
+all dependencies on Operator framework and keeps only the data model of
```

Comment:
I wonder if the "It removes all dependencies on ..." part of the sentence should be removed/toned down. It could be misinterpreted as a negative by non-initiated bystanders.

Reply:
Thanks for catching this, it was meant in the sense that we can't have Operator dependencies since we are writing a reactive interface. But I can see how it could be taken out of context. I'll update it.